core/src/fields.rs

```diff
@@ -469,50 +469,150 @@ impl MongoFields {
     // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
     // timeout, the query must finish before the timeout or an error is returned.
     pub fn list_columns(
-        _mongo_connection: &MongoConnection,
+        mongo_connection: &MongoConnection,
         _query_timeout: Option<i32>,
-        db_name_filter: &str,
-        collection_name_filter: &str,
+        db_name: Option<&str>,
+        collection_name_filter: Option<&str>,
+        field_name_filter: Option<&str>,
     ) -> Self {
+        let dbs = db_name.map_or_else(
```

Comment:
I can't ;)

The "filter" has to be a single name, that's what's happening here. It can also be null, which means all catalogs. You can see this in their code example:

https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function?view=sql-server-ver16

It's unfortunate that the code example seems to be the only thing showing the NULL means "all catalogs".

Reply:
umh, what I read in the spec is that `CatalogName` is either a specific catalog name or an empty string in which case it means tables with no catalogs. But since I don't think MongoDB let you have a collection outside a database, an empty catalog name should always return an empty resulset, no?
Did I miss something?

> CatalogName
[Input] Catalog name. If a driver supports catalogs for some tables but not for others, such as when the driver retrieves data from different DBMSs, an empty string ("") indicates those tables _**that do not have catalogs**_. CatalogName cannot contain a string search pattern.